packages/in_app_purchase/in_app_purchase_android/android/src/main/java/io/flutter/plugins/inapppurchase/MethodCallHandlerImpl.java

```diff
@@ -315,13 +323,23 @@ private void queryPurchasesAsync(String skuType, MethodChannel.Result result) {
           @Override
           public void onQueryPurchasesResponse(
               BillingResult billingResult, List<Purchase> purchasesList) {
-            final Map<String, Object> serialized = new HashMap<>();
-            // The response code is no longer passed, as part of billing 4.0, so we pass OK here
-            // as success is implied by calling this callback.
-            serialized.put("responseCode", BillingClient.BillingResponseCode.OK);
-            serialized.put("billingResult", Translator.fromBillingResult(billingResult));
-            serialized.put("purchaseList", fromPurchasesList(purchasesList));
-            result.success(serialized);
+            handler.post(
+                new Runnable() {
+                  public void run() {
+                    final Map<String, Object> serialized = new HashMap<>();
+                    if (billingResult.getResponseCode() == BillingClient.BillingResponseCode.OK) {
+                      serialized.put("responseCode", billingResult.getResponseCode());
+                      serialized.put("billingResult", Translator.fromBillingResult(billingResult));
+                      serialized.put("purchaseList", fromPurchasesList(purchasesList));
+                      result.success(serialized);
+                    } else {
+                      result.error(
```

Comment:
If all we're doing in the callback is calling `result`, why do we need to post back to the main thread? I thought this was threadsafe now on mobile after Aaron's background platform channel work.

Reply:
Just got back from vacation.  Yea, posting to the platform thread is no longer necessary.  This appears to be an unnecessary hop.